docs/adrs/003-Retry_endpoint_design.md

```diff
@@ -35,26 +35,26 @@ The endpoint takes a job or list of job ids and then attempts to resubmit them t
 * EE2 looks up the job versions and parameters, and then submits the job to be retried, incrementing the `retry_count`
   of the job being retried, and the newly launched job gains a pointer to the `_PARENT_RETRY_JOB_ID`
 * The job is submitted and upon successful submission, notifies the `retry_parent` and notifies the `parent_job_id` that a new `child_job` has been added
+* Jobs submitted by an administrator with a ResourceRequirements use resource requirements from the catalog / ee2 config exclusively
 
-
-### Batch Behavior
-* If a job has the attribute of `batch_job=True` the retry will fail, since there is no method to re-run. This is a bug, as it doesn't fail gracefully.
-* If a job has the attribute of `batch_job=True`, but is actually a child job, the parent will be notified of this new retried job
-* Multiple in-flight retries are allowed.
+### Batch Jobs Behavior
+* Adds child_job_id to parent_job_id.child_job_ids
 
 ## Retry_job behavior
-* Blocking and single submit to HTCondor. It should be fine
+* Blocking and single submit to HTCondor. It should be fine as it returns relatively quickly
   
 ## Retry_jobs behavior
-* Submitting multiple jobs uses the `run_job` endpoint, and is blocking (NOT OK!)
+* Submitting multiple jobs uses the `run_job` endpoint, and is blocking. This can cause issues if the network drops, and makes the narrative not aware of the state of the retry. Submitting 100 jobs currently takes 12 seconds, and that is a lot of time for things to go wrong. 
```

Comment:
Yes. Run_job is the endpoint

Reply:
I'm confused by this answer - the retry endpoint is `retry_jobs` surely? https://github.com/kbase/execution_engine2/blob/develop/execution_engine2.spec#L247

Reply:
Yes, the retry endpoint is used for retrying jobs. The retry endpoint doesn't submit jobs though, it uses the path taken by `run_job` (aka run_job function, as opposed to using the run_jobs function) in order to create a record in mongo and submit to htcondor though

Reply:
Yeah, that looks like a typo - should read "Submitting multiple jobs uses the `retry_jobs` endpoint", right?

Reply:
Hopefully the making the narrative aware of the state of the retry will be mitigated by the narrative backend. It just blocks on the call anyway, with a minute timeout.

Reply:
As long as the user doesn't kill the kernel at that time, all should be well. Of course, if it were me, and it looked frozen for more than a couple minutes, I'd probably restart.

Reply:
>  The retry endpoint doesn't submit jobs though, it uses the path taken by run_job
But it doesn't, which is why I'm confused. The call stack is
```
retry_jobs -> retry_multiple -> _retry -> run -> _run
```
the call there calls methods on `Condor`, so `run_job` is never in the call stack

Reply:
The path by run_job to submit to condor begins with `run(params=params, as_admin=as_admin)`
The path by retry to submit condor begins with `retry_job_id = self.run(params=run_job_params, as_admin=as_admin)`

They aren't calling run_job_batch, for example.

Reply:
> I never mentioned a call stack, and I clarified in the comment that its using the path taken by run_job.. What do you want to sentence to say instead?

I don't currently want it to say anything in particular because I don't really understand the point that it's trying to make, and when I read it it looks incorrect, e.g.:
```
## Retry_jobs behavior
Submitting multiple jobs uses the `run_job` endpoint, and is blocking
```
That makes it sound like `retry_jobs` calls `run_job` under the hood, which it doesn't - that's what I was talking about when I posted the stack.

How about I back up and just ask this - what is the message that the reader of that sentence should get?

Reply:
The message of the reader of the sentence should get is 

Job submission for multiple jobs uses the same business logic as the run_job endpoint and thus is currently blocking. run_jobs_batch is not used.

Reply:
`run_jobs_batch` is blocking too, right? That seems to imply that it isn't.

Reply:
Ok, I would suggest something like
```
Submitting multiple jobs for retry serially calls the same code path used for running a single job and blocks until all jobs have been submitted to the condor queue. This can cause issues if the network drops, and makes the narrative not aware of the state of the retry. Submitting 100 jobs currently takes 12 seconds, and that is a lot of time for things to go wrong. 
```
Does that sound right?

Reply:
Updated it

Reply:
:+1: